docs/excel/excel-add-ins-pivots.md

```diff
@@ -0,0 +1,216 @@
+In the example, both **Farm** and **Type** are rows, with the crate sales as the data. 
+
+![A pivot table showing the total sales of different fruit based on the farm they came from.](../images/excel-pivots-data-hierarchy.png)
+
+```ts
+await Excel.run(async (context) => {
+	let pivotTable = context.workbook.worksheets.getActiveWorksheet().pivotTables.getItem("Farm Sales");
+
+	pivotTable.rowHierarchies.add(pivotTable.hierarchies.getItem("Farm"));
+	pivotTable.rowHierarchies.add(pivotTable.hierarchies.getItem("Type"));
+
+	pivotTable.dataHierarchies.add(pivotTable.hierarchies.getItem("Crates Sold at Farm"));
+	pivotTable.dataHierarchies.add(pivotTable.hierarchies.getItem("Crates Sold Wholesale"));
+
+	await context.sync();
+});
+```
+
+## Change aggregation function
+
+Data hierarchies have their values aggregated into a sum by default. The `summarizeBy` parameter defines this behavior based on an `AggregrationFunction` type. 
+
+## Pivot table layouts
+
+A pivot table layout defines the placement of hierarchies and their data. You access the layout to determine the ranges where data is stored. 
+The following code demonstrates how to get the last row from a pivot table by going through the layout.
+
+```ts
+    await Excel.run(async (context) => {
+        let pivotTable = context.workbook.worksheets.getActiveWorksheet().pivotTables.getItem("Farm Sales");
+        let range = pivotTable.layout.getDataBodyRange();
+        let grandTotalRange = range.getLastRow();
+        grandTotalRange.load();
+        await context.sync();
+    });
+```
+
+Pivot tables have three layout styles: Compact, Outline, and Tabular. We’ve seen the compact style in the previous examples. 
+The following examples use the outline and tabular styles, respectively. The code sample shows how to cycle between the different layouts.
+
+### Outline layout
+![A pivot table using the outline layout.](../images/excel-pivots-outline-layout.png)
+
+### Tabular layout
+![A pivot table usingthe tabular layout.](../images/excel-pivots-tabular-layout.png)
+
+```ts
+await Excel.run(async (context) => {
+	let pivotTable = context.workbook.worksheets.getActiveWorksheet().pivotTables.getItem("Farm Sales");
+	pivotTable.layout.load("layoutType");
+	await context.sync();
+	if (pivotTable.layout.layoutType === "Compact") {
+		pivotTable.layout.layoutType = "Outline";
+	} else if (pivotTable.layout.layoutType === "Outline") {
+		pivotTable.layout.layoutType = "Tabular";
+	} else {
+		pivotTable.layout.layoutType = "Compact";
+	}
+	
+	await context.sync();
+});
+```
+
+## Change hierarchy names
+
+Hierarchy fields are editable. The following code demonstrates how to change the displayed names of two data hierarchies.
+
+```ts
+await Excel.run(async (context) => {
+	let dataHierarchies = context.workbook.worksheets.getActiveWorksheet()
+		.pivotTables.getItem("Farm Sales").dataHierarchies;
+	dataHierarchies.load();
```

Comment:
It looks like you only read the `name `property, so you specify that one in the load:
`dataHierarchies.load("name");`